### ISSUE TITLE
Problem in NLU training - insufficient training data or something else ?

### ISSUE THREAD

**Issue by anshulsolanki** [RasaHQ/rasa_core#136]
**Rasa Core version**:
0.7

**Python version**: 
3.6

**Operating system** (windows, osx, ...):
Windows 10

**Issue**:
I am building the restaurantbot example ( https://github.com/RasaHQ/rasa_core/tree/master/examples/restaurantbot). When i do the NLU training , i get warnings
INFO:rasa_nlu.converters:Training data format at data/franken_data.json is rasa_nlu
DEBUG:rasa_nlu.training_data:Validating training data...
INFO:rasa_nlu.training_data:Training data stats: 
	- intent examples: 1954 (6 distinct intents)
	- found intents: 'affirm', 'deny', 'greet', 'inform', 'request_info', 'thankyou'
	- entity examples: 1017 (5 distinct entities)
	- found entities: 'cuisine', 'info', 'location', 'people', 'price'

INFO:rasa_nlu.utils.spacy_utils:Trying to load spacy model with name 'en'
INFO:rasa_nlu.components:Added 'nlp_spacy' to component cache. Key 'nlp_spacy-en'.
INFO:rasa_nlu.model:Starting to train component nlp_spacy
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component tokenizer_spacy
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component intent_featurizer_spacy
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component intent_classifier_sklearn
Fitting 2 folds for each of 6 candidates, totalling 12 fits
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\AppData\Local\Continuum\anaconda3\lib\site-packages\sklearn\metrics\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
[Parallel(n_jobs=1)]: Done  12 out of  12 | elapsed:    6.1s finished
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component ner_crf
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Starting to train component ner_synonyms
INFO:rasa_nlu.model:Finished training component.
INFO:rasa_nlu.model:Successfully saved model into 'C:\work\prj\rasa_restaurant_example\models\nlu\default\current'

As i understand these might be because of less training data for intents. But considering that training data is coming from standard source ( https://core.rasa.ai/tutorial_supervised.html ) , i though it is enough.

Can someone confirm if the restaurantbot example trains well for NLU using data provided on github, or we need to augment this data. 

**Content of domain file** (if used & relevant):
I have used all default files available on github.

**Comment by anshulsolanki** [comment]
Update=> Yes it is training data issue. I removed the intents 'request_info' and 'greet' because they only had 3 examples each , re-trained NLU and warning are gone.

I will try adding more examples for 'request_info' and 'greet' and check back in github.

Anybody already have updated data, please post here.

**Comment by anshulsolanki** [comment]
Closing the issue , since this is related to training data

**Comment by daysm** [comment]
@anshulsolanki I have the same problem, but removing the intents you mentioned (I removed them from the training data and in the domain definition) did not work, I'm still getting the same error messages. Is there anything else you did? Could you share the steps you went through to fix this issue?

**Comment by anshulsolanki** [comment]
I did not remove the intents from domain definition file. It is not needed.
you need to make sure that intent(s) with less examples are either removed or else add more examples for such intents.
for restaurant bot examples, during debugging, I figured that there were not enough examples for training intent or greet and request_info. So I added new examples of greet and removed request_info ( for now) and NLU training is good.

Do not ignore any warning that comes in training 

If you can share the data i can run and look at the error.

**Comment by HenrikGeiger** [comment]
I get the exact same issue with the latest files of the example folder. I tried it with python 3.6 on a windows 7 machine. Now I am not sure what I can do to solve this Problem. Is there anything what I can do?

**Comment by HenrikGeiger** [comment]
I get the exact same issue with the latest files of the example folder. I tried it with python 3.6 on a windows 7 machine. Now I am not sure what I can do to solve this Problem. Is there anything what I can do?

**Comment by rhk217** [comment]
I just added the following examples (6 more request_info examples and 7 more greet examples). Now the errors are not coming up. You can just add these to franken_data.json.
```
      {
        "text": "hey",
        "intent": "greet",
        "entities": []
      },
      {
        "text": "morning",
        "intent": "greet",
        "entities": []
      },
      {
        "text": "hello there",
        "intent": "greet",
        "entities": []
      },
      {
        "text": "hi there",
        "intent": "greet",
        "entities": []
      },
      {
        "text": "good afternoon",
        "intent": "greet",
        "entities": []
      },
      {
        "text": "hey bot",
        "intent": "greet",
        "entities": []
      },
      {
        "text": "good evening",
        "intent": "greet",
        "entities": []
      },
      {
        "text": "whats their phone number",
        "intent": "request_info",
        "entities": [
          {
            "start": 12,
            "end": 24,
            "value": "phone number",
            "entity": "info"
          }
        ]
      },
      {
        "text": "whats their address",
        "intent": "request_info",
        "entities": [
          {
            "start": 12,
            "end": 19,
            "value": "address",
            "entity": "info"
          }
        ]
      },
      {
        "text": "what is the address of the restaurant",
        "intent": "request_info",
        "entities": [
          {
            "start": 12,
            "end": 19,
            "value": "address",
            "entity": "info"
          }
        ]
      },
      {
        "text": "may i have the address of the restaurant",
        "intent": "request_info",
        "entities": [
          {
            "start": 15,
            "end": 22,
            "value": "address",
            "entity": "info"
          }
        ]
      },
      {
        "text": "do you have their phone number",
        "intent": "request_info",
        "entities": [
          {
            "start": 18,
            "end": 30,
            "value": "phone number",
            "entity": "info"
          }
        ]
      },
      {
        "text": "do you have their address",
        "intent": "request_info",
        "entities": [
          {
            "start": 18,
            "end": 25,
            "value": "address",
            "entity": "info"
          }
        ]
      },
```

**Comment by tmbo** [comment]
@rhk217 if you don't mind, can you create a PR for that?
